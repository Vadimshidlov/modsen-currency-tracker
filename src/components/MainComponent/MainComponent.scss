@use "@/constants/styles/fonts" as fonts;
@use "@/constants/styles/colors" as colors;
@import "@/index.scss";

.main-block {
    display: flex;
    justify-content: center;
    padding-top: 57px;
}
