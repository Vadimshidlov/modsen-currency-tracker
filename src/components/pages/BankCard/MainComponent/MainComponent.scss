@use "@/constants/styles/fonts" as fonts;
@use "@/constants/styles/colors" as colors;

.main-block {
    display: flex;
    flex-direction: column;
    justify-content: center;
}
