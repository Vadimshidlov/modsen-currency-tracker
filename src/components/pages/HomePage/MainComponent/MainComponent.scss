@use "@/constants/styles/fonts" as fonts;
@use "@/constants/styles/colors" as colors;
@import "@/index.scss";

.main-block {
    display: flex;
    flex-direction: column;
    justify-content: center;
}
