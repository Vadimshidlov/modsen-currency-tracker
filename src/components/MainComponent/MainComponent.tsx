import React, { useEffect } from "react";
import LastUpdate from "@/components/LastUpdate/LastUpdate";
import "@/components/MainComponent/MainComponent.scss";
// import CurrencyCard from "@/components/CurrencyCard/CurrencyCard";
import CurrencyStocks from "@/components/CurrencyStocks/CurrencyStocks";
import CurrencyQoutes from "@/components/CurrencyQoutes/CurrencyQoutes";
import { useActions } from "@/store/hooks/useActions";

export default function MainComponent() {
    const { getCurrency } = useActions();

    useEffect(() => {
        getCurrency();
    }, [getCurrency]);

    return (
        <div className="main-block">
            <LastUpdate />
            <CurrencyStocks />
            <CurrencyQoutes />
            {/* <CurrencyCard /> */}
        </div>
    );
}
