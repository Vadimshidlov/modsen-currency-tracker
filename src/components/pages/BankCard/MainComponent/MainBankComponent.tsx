import React, { Component } from "react";
import { connect } from "react-redux";
import LastUpdate from "@/components/pages/TimelinePage/LastUpdate/LastUpdate";
import "@/components/pages/HomePage/MainComponent/MainComponent.scss";
import { getCurrency } from "@/store/action-creators/getCurrency";
import { getMapData } from "@/store/action-creators/getMapData";

import MapController from "@/components/pages/BankCard/MapController/MapController";
import { MainBankComponentPropsType } from "@/types/BankCardPageTypes/types";

class MainBankComponent extends Component<MainBankComponentPropsType> {
    componentDidMount() {
        const { getCurrency: getCurrencyData, getMapData: getAllMapData } = this.props;

        getCurrencyData();
        getAllMapData();
    }

    componentDidUpdate() {
        const { getCurrency: getCurrencyData } = this.props;

        getCurrencyData();
    }

    render() {
        return (
            <main className="main-block">
                <LastUpdate />
                <MapController />
            </main>
        );
    }
}

export default connect(null, { getCurrency, getMapData })(MainBankComponent);
