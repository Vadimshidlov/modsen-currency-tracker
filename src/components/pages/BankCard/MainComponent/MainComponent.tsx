import React, { Component } from "react";
import { connect } from "react-redux";
import LastUpdate from "@/components/pages/TimelinePage/LastUpdate/LastUpdate";
import "@/components/pages/HomePage/MainComponent/MainComponent.scss";
import { getCurrency } from "@/store/action-creators/getCurrency";
import MapController from "@/components/pages/BankCard/MapController/MapController";

export type MainComponentPropsType = {
    getCurrency: () => Promise<void>;
};

class MainComponent extends Component<MainComponentPropsType> {
    componentDidMount() {
        const { getCurrency: getCurrencyData } = this.props;

        getCurrencyData();
    }

    componentDidUpdate() {
        const { getCurrency: getCurrencyData } = this.props;

        getCurrencyData();
    }

    render() {
        return (
            <main className="main-block">
                <LastUpdate />
                <MapController />
            </main>
        );
    }
}

export default connect(null, { getCurrency })(MainComponent);
