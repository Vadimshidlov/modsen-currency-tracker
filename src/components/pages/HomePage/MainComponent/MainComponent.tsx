import React, { useEffect } from "react";
import LastUpdate from "@/components/pages/HomePage/LastUpdate/LastUpdate";
import "@/components/pages/HomePage/MainComponent/MainComponent.scss";
import CurrencyStocks from "@/components/pages/HomePage/CurrencyStocks/CurrencyStocks";
import CurrencyQoutes from "@/components/pages/HomePage/CurrencyQoutes/CurrencyQoutes";
import { useActions } from "@/store/hooks/useActions";

export default function MainComponent() {
    const { getCurrency } = useActions();

    useEffect(() => {
        getCurrency();
    }, [getCurrency]);

    return (
        <main className="main-block main-block__container">
            <LastUpdate />
            <CurrencyStocks />
            <CurrencyQoutes />
        </main>
    );
}
