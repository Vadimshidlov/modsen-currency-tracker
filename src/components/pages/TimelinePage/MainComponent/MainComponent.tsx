import React, { Component } from "react";
import { connect } from "react-redux";
import LastUpdate from "@/components/pages/TimelinePage/LastUpdate/LastUpdate";
import "@/components/pages/HomePage/MainComponent/MainComponent.scss";
import { getCurrency } from "@/store/action-creators/getCurrency";
import CurrencyChartController from "@/components/pages/TimelinePage/CurrencyChartController/CurrencyChartController";

export type MainComponentPropsType = {
    getCurrency: () => Promise<void>;
};

class MainComponent extends Component<MainComponentPropsType> {
    componentDidMount() {
        const { getCurrency: getCurrencyData } = this.props;

        getCurrencyData();
    }

    componentDidUpdate() {
        const { getCurrency: getCurrencyData } = this.props;

        getCurrencyData();
    }

    render() {
        return (
            <main className="main-block main-block__container">
                <LastUpdate />
                <CurrencyChartController />
            </main>
        );
    }
}

export default connect(null, { getCurrency })(MainComponent);
