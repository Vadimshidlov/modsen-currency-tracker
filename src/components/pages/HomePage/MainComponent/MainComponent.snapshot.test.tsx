import "@testing-library/jest-dom";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import { BrowserRouter as Router } from "react-router-dom";
import MainComponent from "@/components/pages/HomePage/MainComponent/MainComponent";
import { store } from "@/store";

describe("MainComponent component snapshot tests", () => {
    it("should match snapshot", () => {
        jest.clearAllMocks();

        const tree = renderer
            .create(
                <Provider store={store}>
                    <Router>
                        <MainComponent />
                    </Router>
                </Provider>,
            )
            .toJSON();

        expect(tree).toMatchSnapshot();
    });
});
