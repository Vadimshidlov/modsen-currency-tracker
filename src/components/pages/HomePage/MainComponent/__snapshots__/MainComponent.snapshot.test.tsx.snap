// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainComponent component snapshot tests should match snapshot 1`] = `
<main
  className="main-block main-block__container"
>
  <div
    className="last-update"
  >
    <div
      className="pulsating-circle"
    />
    <p
      className="last-update__text"
    >
      Last updated 
    </p>
  </div>
  <div
    className="stocks__container"
  >
    <p
      className="stocks__title title"
    >
      Stocks
    </p>
    <div
      className="title__border"
    />
    <div
      className="stocks__items"
    >
      <button
        className="currency-card__container"
        disabled={true}
      >
        <svg
          className="currency-card__icon"
        />
        <div
          className="currency-card__info"
        >
          <p
            className="currency-card__titile"
          >
            Bovespa Index
          </p>
          <p
            className="currency-card__percent"
          >
            0.15%
          </p>
        </div>
      </button>
      <button
        className="currency-card__container"
        disabled={true}
      >
        <svg
          className="currency-card__icon"
        />
        <div
          className="currency-card__info"
        >
          <p
            className="currency-card__titile"
          >
            IFIX
          </p>
          <p
            className="currency-card__percent"
          >
            0.15%
          </p>
        </div>
      </button>
    </div>
  </div>
  <div
    className="quotes__container"
  >
    <p
      className="quotes__title title"
    >
      Quotes
    </p>
    <div
      className="title__border"
    />
    <div
      className="quotes__items"
    />
  </div>
</main>
`;
